--- src/command.C	2025-11-30 11:15:04.000000000 +0100
+++ src/command.C	2026-06-08 13:00:00.000000000 +0200
@@ -1045,10 +1045,7 @@
       want_refresh = 1;
     }
 
-  if (option (Opt_cursorBlink) || (priv_modes & PrivMode_BlinkingCursor))
-    cursor_blink_ev.again ();
-  else
-    cursor_blink_ev.stop ();
+  cursor_blink_ev.stop ();
 }
 
 void
@@ -1750,8 +1747,7 @@
         }
 #endif
 #if CURSOR_BLINK
-      if (option (Opt_cursorBlink))
-        cursor_blink_ev.again ();
+      cursor_blink_reset ();
 #endif
 #if OFF_FOCUS_FADING
       if (rs[Rs_fade])
